### ISSUE TITLE
Cant push to private repository

### ISSUE THREAD

**Issue by barbarosh** [jenkinsci/docker-plugin#785]
- [ ] docker-plugin 1.2.0
 - [ ] docker-java-api 3.1.5.1
 - [ ] jenkins 2.229
 - [ ] docker Version = 18.09.7, API Version = 1.39

Problem: Cant push to private repository

```
FATAL: com/google/common/io/BaseEncoding
java.lang.NoClassDefFoundError: com/google/common/io/BaseEncoding
	at com.github.dockerjava.core.exec.AbstrDockerCmdExec.registryAuth(AbstrDockerCmdExec.java:45)
	at com.github.dockerjava.core.exec.AbstrDockerCmdExec.resourceWithAuthConfig(AbstrDockerCmdExec.java:77)
	at com.github.dockerjava.core.exec.PushImageCmdExec.execute0(PushImageCmdExec.java:39)
	at com.github.dockerjava.core.exec.PushImageCmdExec.execute0(PushImageCmdExec.java:16)
	at com.github.dockerjava.core.exec.AbstrAsyncDockerCmdExec.execute(AbstrAsyncDockerCmdExec.java:56)
	at com.github.dockerjava.core.exec.AbstrAsyncDockerCmdExec.exec(AbstrAsyncDockerCmdExec.java:21)
	at com.github.dockerjava.core.exec.AbstrAsyncDockerCmdExec.exec(AbstrAsyncDockerCmdExec.java:12)
	at com.github.dockerjava.core.command.AbstrAsyncDockerCmd.exec(AbstrAsyncDockerCmd.java:21)
	at com.nirima.jenkins.plugins.docker.builder.DockerBuilderPublisher$Run.pushImages(DockerBuilderPublisher.java:433)
	at com.nirima.jenkins.plugins.docker.builder.DockerBuilderPublisher$Run.run(DockerBuilderPublisher.java:330)
	at com.nirima.jenkins.plugins.docker.builder.DockerBuilderPublisher.perform(DockerBuilderPublisher.java:463)
	at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:78)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:741)
	at hudson.model.Build$BuildExecution.build(Build.java:206)
	at hudson.model.Build$BuildExecution.doRun(Build.java:163)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:504)
	at hudson.model.Run.execute(Run.java:1880)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:97)
	at hudson.model.Executor.run(Executor.java:428)
```

**Comment by pjdarton** [comment]
Looks like the same cause as #784

**Comment by pjdarton** [comment]
FYI this (and #784) should be fixed by upgrading to docker-java-api-plugin version 3.1.5.2
A pre-release of this can be located via https://github.com/jenkinsci/docker-plugin/issues/784#issuecomment-609038432

@barbarosh Can you download the pre-release and verify that this fixes the issue for you?
(speedy feedback like this will speed up an official fix...)

**Comment by ericcitaire** [comment]
I tested the pre-release with the following code :

```groovy
pipeline {
    agent any

    stages {
        stage('Hello') {
            steps {
                echo 'Hello World'
                writeFile file: 'Dockerfile', text: 'FROM jenkins/jnlp-agent-docker'
                step([
                    $class: 'DockerBuilderPublisher',
                    dockerFileDirectory: '',
                    tagsString: "my/repo:${env.BUILD_ID}",
                    pushCredentialsId: 'dockerhub',
                    pushOnSuccess: true
                ])
            }
        }
    }
}

```

The image is built and pushed, properly tagged.

**Comment by pjdarton** [comment]
Given that we _think_ we've fixed it (as part of fixing #784) and we've no reason to suspect we're wrong, I'm going to close this.

@barbarosh - if docker-plugin version 1.2.0 and docker-java-api-plugin version 3.1.5.2 doesn't work for you, feel free to re-open this issue and provide further details.

**Comment by jrichmond4** [comment]
I ran into this issue today with docker-plugin version 1.2.9 and docker-api-plugin version 3.2.13-37.vf3411c9828b9
